Libs/MRML/Core/vtkMRMLSliceNode.cxx

```diff
@@ -636,47 +794,17 @@ void vtkMRMLSliceNode::UpdateMatrices()
       this->Modified();
       }
 
-    const char *orientationString = "Reformat";
-    if ( this->SliceToRAS->GetElement(0, 0) == -1.0 &&
-         this->SliceToRAS->GetElement(1, 0) ==  0.0 &&
-         this->SliceToRAS->GetElement(2, 0) ==  0.0 &&
-         this->SliceToRAS->GetElement(0, 1) ==  0.0 &&
-         this->SliceToRAS->GetElement(1, 1) ==  1.0 &&
-         this->SliceToRAS->GetElement(2, 1) ==  0.0 &&
-         this->SliceToRAS->GetElement(0, 2) ==  0.0 &&
-         this->SliceToRAS->GetElement(1, 2) ==  0.0 &&
-         this->SliceToRAS->GetElement(2, 2) ==  1.0 )
-      {
-        orientationString = "Axial";
-      }
-
-    if ( this->SliceToRAS->GetElement(0, 0) ==  0.0 &&
-         this->SliceToRAS->GetElement(1, 0) == -1.0 &&
-         this->SliceToRAS->GetElement(2, 0) ==  0.0 &&
-         this->SliceToRAS->GetElement(0, 1) ==  0.0 &&
-         this->SliceToRAS->GetElement(1, 1) ==  0.0 &&
-         this->SliceToRAS->GetElement(2, 1) ==  1.0 &&
-         this->SliceToRAS->GetElement(0, 2) ==  1.0 &&
-         this->SliceToRAS->GetElement(1, 2) ==  0.0 &&
-         this->SliceToRAS->GetElement(2, 2) ==  0.0 )
+    // It update the Orientation Matrices with the Position coordinates
+    // which are set only on the SliceToRAS matrix.
+    std::map<vtkStdString, vtkMatrix4x4*>::iterator it;
+    for (it = this->OrientationMatrices.begin(); it != this->OrientationMatrices.end(); ++it)
       {
-        orientationString = "Sagittal";
+      it->second->SetElement(0, 3, this->SliceToRAS->GetElement(0, 3));
```

Comment:
Here I just add, to the presets also the "Origin" information stored in the SliceToRAS. 
This because, when SetOrietation is called, the SliceToRAS matrix is override by the present (all the 4x4 elements)